docs/rfcs/003-interactive-log-view.md

```diff
@@ -0,0 +1,63 @@
+# Interactive Log View
+
+## Status
+
+Proposed
+
+## Summary
+
+Open a pane item displaying a navigable, live 3D model of your git commit graph.
+
+## Motivation
+
+Visualizing the commit graph is an important tool for learning git. A live view of the commit graph can inobtrusively help git newcomers form the correct mental model for commit graph operations (branching, merging, rebasing).
+
+We do not currently have a way to initiate merge, rebase, or cherry-pick operations from the editor. A log view would be the most logical place to do so, because it's where you have the most context about the existing relationship of the refs you'd like to merge or rebase.
+
+A log view would grant us a way to answer the question of where you are relative to nearby refs and tags.
+
+Why 3D? Flat visualizations of git history quickly fall victim to the "guitar hero problem":
+
+[![Git guitar hero](https://user-images.githubusercontent.com/17565/39223169-6f5d6b1e-480e-11e8-809b-60d1dc8261fb.png)](https://twitter.com/henryhoffman/status/694184106440200192)
+
+I'm hopeful that the extra "space" afforded by rendering in WebGL and level-of-detail fading can give us a view that remains functional even in the face of dramatic tangles.
+
+## Explanation
+
+The log view is a pane item launched by:
+
+* The `github:log` command in the command palette.
+* An "Open Log" menu item in the context menu on the recent commit history.
+
+When launched, the log view pane uses WebGL to render a 3D model of your commit graph.
+
+* Each commit is rendered as a sphere, possibly with adjacent text (see below).
+* A line joins each commit to its parent.
+* Refs are rendered as rectangular tags attached to the relevant commits.
+* When a remote is present, commits that are not reachable from any remote ref are rendered _[TODO]_. Commits that have been fetched but are not reachable from any local ref are rendered _[TODO]_.
```

Comment:
There's basically 2 pieces of information you want to convey (preferably visually and at a glance) here:
- Local, remote, or both
- which remote(s)

My initial thought (assuming this view has a strong sense of up and down) is that local only commits are rendered as a pyramid (pointing up, which mildly indicates push). Remote only commits are rendered as an inverted pyramid (pointing down, which mildly indicates pull). Any commit that is both local and remote will be a sphere.

My idea for indicating which remote/remotes a commit is part of is to render an icon for each remote a commit is part of. We could use a house icon for origin, and a wavy (diagonal upwards) icon for upstream. Maybe basic geometric shapes (square, circle, triangle, star) for any other remotes. 

If you give each remote its own color, then you could mix the colors to determine the color of the sphere/pyramid. Maybe local only is gray, origin is green, upstream is blue, etc. Then upstream and origin would be cyan/teal (mix of blue and green), etc.

Reply:
This scheme works well with the levels of detail. Highest level of detail lets you hover the remote icons to see the names for the remotes. Medium level of detail lets you see the icons, but maybe smaller, or maybe only the first remote. Low level of detail has no icons, but retains the shape and color, so you can visually see if there's a "far away" commit that hasn't been pushed yet.